src/SmartTransactionsController.ts

```diff
@@ -289,7 +350,7 @@ export default class SmartTransactionsController extends BaseController<
               .slice(0, cancelledNonceIndex)
               .concat(currentSmartTransactions.slice(cancelledNonceIndex + 1))
               .concat(historifiedSmartTransaction)
-          : currentSmartTransactions.concat(historifiedSmartTransaction);
+          : currentSmartTransactions?.concat(historifiedSmartTransaction);
```

Comment:
`nextSmartTransactions` would be undefined in state rather than an empty array if  `currentSmartTransactions` is undefined. Is that what we want?

Reply:
nope 🤔 also not sure why this change was made...

Reply:
done here: https://github.com/MetaMask/smart-transactions-controller/pull/237/commits/caf5ac0a59b73823ef6977cb480bf4c3ec2b5b22